src/explorers/ethereum/etherscan.js

```diff
@@ -51,66 +33,57 @@ function parseEtherScanResponse (jsonResponse, block) {
   return generateTransactionData(opReturnScript, issuingAddress, date, undefined);
 }
 
-function getEtherScanBlock (jsonResponse, chain) {
+async function getEtherScanBlock (jsonResponse, chain) {
   const data = jsonResponse.result;
   const blockNumber = data.blockNumber;
-  const action = '&action=eth_getBlockByNumber&boolean=true&tag=';
-  let etherScanUrl;
-  if (chain === BLOCKCHAINS.ethmain.code) {
-    etherScanUrl = TRANSACTIONS_APIS_URLS[TRANSACTION_APIS.Etherscan].main + action + blockNumber;
-  } else {
-    etherScanUrl = TRANSACTIONS_APIS_URLS[TRANSACTION_APIS.Etherscan].test + action + blockNumber;
+  const requestUrl = buildTransactionServiceUrl({
+    serviceUrls: getBlockByNumberServiceUrls,
+    transactionIdPlaceholder: TRANSACTION_ID_PLACEHOLDER,
+    transactionId: blockNumber,
+    testApi: isTestChain(chain)
+  });
+
+  try {
+    const response = await request({ url: requestUrl });
+    const responseData = JSON.parse(response);
+    const blockData = responseData.result;
+
+    await checkEtherScanConfirmations(chain, blockNumber);
+    return blockData;
+  } catch (err) {
+    throw new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash'));
   }
+}
 
-  return new Promise((resolve, reject) => {
-    return request({ url: etherScanUrl })
-      .then(function (response) {
-        const responseData = JSON.parse(response);
-        const blockData = responseData.result;
-        try {
-          const checkConfirmationsFetcher = checkEtherScanConfirmations(chain, blockNumber);
-          checkConfirmationsFetcher
-            .then(function () {
-              resolve(blockData);
-            })
-            .catch(function () {
-              reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-            });
-        } catch (err) {
-          // don't need to wrap this exception
-          reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-        }
-      }).catch(function () {
-        reject(new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash')));
-      });
+async function checkEtherScanConfirmations (chain, blockNumber) {
+  const requestUrl = buildTransactionServiceUrl({
+    serviceUrls: getBlockNumberServiceUrls,
+    testApi: isTestChain(chain)
```

Comment:
I think the `testAPI` property should better reflect that it's a boolean. `isTestAPI`?

Reply:
👍